fvm/src/call_manager/default.rs

```diff
@@ -86,6 +86,18 @@ where
     where
         K: Kernel<CallManager = Self>,
     {
+        // We check _then_ set because we don't count the top call. By example:
+        //
+        // 1. If the max depth is 0, call_stack_depth will be 1 and the top-level message won't be
+        //    able to make sub-calls (1 > 0).
+        // 2. If the max depth is 1, the call_stack_depth will be 1 in the top-level message, 2 in
+        //    sub-calls, and said sub-calls will not be able to make further subcalls (2 > 1).
+        //
+        // NOTE: This means we can actually have a full "depth" of max_call_depth + 1, but that's
+        // how the network currently operates.
+        //
+        // NOTE: While lotus adds then checks (unlike the FVM which checks then adds), the
+        // call_stack_depth starts at 0 for the first-level while it's 1 in the FVM.
```

Comment:
It's great to document this, but I find it a bit confusing. Because this is very much a correctness thing, I suspect some formality would help. Maybe:
- define terminology, e.g. top-level message, first-level message, top-call
- define implementation behaviours: what Lotus does and when, what FVM does and when
- define network behaviour pre-FVM and post-FVM
- elaborate on tradeoffs

Reply:
Is the new comment clearer?

The underlying tradaeoff is really: the code is different and the lotus model doesn't work here for "reasons". The goal of this comment is to explain how the FVM's version is correct (or, at least, matches lotus).